### PR TITLE
test framework: deploy control planes in parallel

### PR DESCRIPTION
**Pull request by stevenctl**
We will be running istio in multiple clusters as part most integration tests eventually as part of https://github.com/istio/istio/issues/23618. To speed things up we should be doing as much set-up as possible in parallel. 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: prow/lib.sh [hunk 1]
```diff
@@ -197,13 +197,16 @@ function setup_kind_clusters() {
 
   KUBECONFIG_DIR="$(mktemp -d)"
 
+  docker network create kind
```
Comment (howardjohn) [inline prow/lib.sh:200]: nit: can we only create this if not present in case people run locally and already have this?

**Comment by stevenctl** [comment]
integ-multicluster-k8s-tests_istio passes after adding https://github.com/istio/istio/pull/24523/commits/31369a4d5b6c908872862930bad1e512e5734541 which proves using `GetRootCmd().Execute()` from `istioctl/cmd` doesn't work in concurrent goroutines. Either need to find a good workaround or fix whatever concurrency issues are in that branch of code.

File: istioctl/pkg/install/verify.go [hunk 2]
```diff
@@ -248,7 +247,7 @@ func NewVerifyCommand() *cobra.Command {
 		istioNamespace string
 		opts           clioptions.ControlPlaneOptions
 	)
-	verifyInstallCmd = &cobra.Command{
+	verifyInstallCmd := &cobra.Command{
```
Comment (stevenctl) [inline istioctl/pkg/install/verify.go:250]: having a package level var was unnecessary and caused a panic when run in parallel

File: operator/pkg/helmreconciler/reconciler.go [hunk 3]
```diff
@@ -374,7 +374,8 @@ func (h *HelmReconciler) getCRHash(componentName string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return strings.Join([]string{crName, crNamespace, componentName}, "-"), nil
+
+	return strings.Join([]string{crName, crNamespace, componentName, h.restConfig.Host}, "-"), nil
```
Comment (stevenctl) [inline operator/pkg/helmreconciler/reconciler.go:378]: Open to suggestions other than `h.restConfig.Host` but the goal was to make the cache key unique per-cluster to allow concurrent use.

**Review by howardjohn — APPROVED** [review]

**Review by esnible — APPROVED** [review]
(istioctl changes are the only thing I looked at)

**Pull request merged**